src/Network/Algorand/Api/Indexer.hs

```diff
@@ -313,3 +296,38 @@ data BlockResp = BlockResp
 
 $(deriveJSON algorandTrainOptions 'IdxAccountResponse)
 $(deriveJSON algorandTrainOptions 'BlockResp)
+
+-- | Indexer health information.
+-- From: https://indexer.testnet.algoexplorerapi.io/health
+data Health = Health
+  { hDbAvailable :: Bool
+  , hGenesisHash :: Text
+  , hGenesisId :: Text
+  , hIsMigrating :: Bool
+  , hMessage :: Text
+  , hRound :: Int64
```

Comment:
We have a type for round, please - use `Data.Algorand.Round (Round)`

Reply:
> Since we have it again and we can get network id for indexer - it will be good to add network check at our client functions (but I can do it in separate commit in context of Serokell internal dev)

I see, I added the check according to the old change you linked in the `connectToNode` now. Let me know if this is what you have in mind.

> We have a type for round
Updated this now.

Reply:
> I see, I added the check according to the old change you linked in the connectToNode now. Let me know if this is what you have in mind.

more or less...
a few comments on this:
1. it'll be better to put such checks in separate commit
2. for indexer client we can do same now, isn't it?
3. did you checked that `halgo` (command line tool) works well after your changes?

Reply:
Make sense, I separate the commits now, also add the check for indexer. Testing with `halgo` work fines with both indexer and node.